Lua/Lua.sublime-syntax

```diff
@@ -757,6 +814,442 @@ contexts:
+      push:
+        - pattern-quoted-set-range-end
+        - pattern-quoted-set-range-begin
+
+  pattern-quoted-set-range-begin:
+    - meta_include_prototype: false
+    - match: '-'
+      scope: punctuation.separator.sequence.lua
+      pop: 1
+    - include: string-content
+
+  pattern-quoted-set-range-end:
+    - meta_include_prototype: false
+    - meta_scope: constant.other.range.lua
+    - include: string-escape-pop
+    - match: .
+      pop: 1
+
+  pattern-unquoted:
+    - meta_include_prototype: false
+    - include: pattern-shared
+    - match: '\('
+      scope: punctuation.section.group.begin.lua
+      push:
+        - pattern-shared-norepetition
+        - pattern-unquoted-nested
+        - pattern-shared-norepetition
+    - match: '(\[)(\^)?\]?'
+      captures:
+        1: punctuation.definition.set.begin.lua
+        2: keyword.operator.logical.lua
+      push: pattern-unquoted-set-body
+    - include: pattern-unquoted-charclass
+
+  pattern-unquoted-charclass:
+    - meta_include_prototype: false
+    - include: pattern-shared-charclass
+    - match: \%b
+      scope: keyword.control.balanced.lua
+      push:
+        - pattern-unquoted-balanced-char
+        - pattern-unquoted-balanced-char
+    - match: \%
+      scope: constant.character.escape.lua
+      push:
+        - meta_include_prototype: false
+        - match: .
+          scope: constant.character.escape.lua
+          pop: 1
+
+  pattern-unquoted-balanced-char:
+    - meta_include_prototype: false
+    - match: .
+      scope: keyword.control.balanced.lua
+      pop: 1
+
+  pattern-unquoted-nested:
+    - meta_include_prototype: false
+    - meta_scope: meta.group.lua meta.mode.basic.lua
+    - match: '\)'
+      scope: punctuation.section.group.end.lua
+      pop: 1
+    - include: pattern-unquoted
+
+  pattern-unquoted-set-body:
+    - meta_include_prototype: false
+    - meta_scope: meta.set.lua
+    - match: '\]'
+      scope: punctuation.definition.set.end.lua
+      pop: 1
+    - include: pattern-unquoted-set-ranges
+    - include: pattern-unquoted-charclass
+
+  pattern-unquoted-set-ranges:
+    - meta_include_prototype: false
+    # These only work with direct characters. Lua doesn't support percent escapes in sets
+    - match: (?=[^%\]\-]-[^\]])
+      push:
+        - pattern-unquoted-set-range-end
+        - pattern-unquoted-set-range-begin
+
+  pattern-unquoted-set-range-begin:
+    - meta_include_prototype: false
+    - match: '-'
+      scope: punctuation.separator.sequence.lua
+      pop: 1
+
+  pattern-unquoted-set-range-end:
+    - meta_include_prototype: false
+    - meta_scope: constant.other.range.lua
+    - match: .
+      pop: 1
+
+  # this gets included in expressions that want to have a pattern string
+  pattern-string:
+    - match: \[(=*)\[
+      scope: punctuation.definition.string.begin.lua
+      set:
+        - pattern-multiline-string-body
+        - pattern-test-first-caret-multiline
```

Comment:
To reliably escape on proper amount of `=` chars, embed in combo with pop could be used here:

```suggestion
    - match: \[(=*)\[
      scope: meta.string.lua string.quoted.multiline.lua punctuation.definition.string.begin.lua
      embed: pattern-multiline-string-body
      embed_scope: meta.string.lua string.quoted.multiline.lua
      escape: \]\1\]
      escape_captures:
        0: meta.string.lua string.quoted.multiline.lua punctuation.definition.string.end.lua
      pop: 1
```